perfmetrics/scripts/README.md

```diff
@@ -0,0 +1,82 @@
+```
+
+## How to run
+1. Create a GCP VM with OS version as Ubuntu 20.04. Follow this [documentation](https://cloud.google.com/compute/docs/create-linux-vm-instance) and start your VM. Follow the next steps in your VM.
+2. Install the required packages as mentioned in the above section
+3. Clone the GCSFuse repo and cd into the perfmetrics/scripts directory:
+```bash
+git clone https://github.com/GoogleCloudPlatform/gcsfuse.git
+cd gcsfuse/perfmetrics/scripts
+```
+4. Create a directory and mount your GCS bucket into it using GCSFuse:
+```bash
+mkdir -p your-directory-name
+GCSFUSE_FLAGS="--implicit-dirs --max-conns-per-host 100 --disable-http2"
+BUCKET_NAME=your-bucket-name
+MOUNT_POINT=your-directory-name
+gcsfuse $GCSFUSE_FLAGS $BUCKET_NAME $MOUNT_POINT
+```
+
+5. Create a FIO job file `your-job-file.fio` according to [this format](https://fio.readthedocs.io/en/latest/fio_doc.html#job-file-format) and place it in the `job_files` directory. Note that you can use only one job file. However, you can put any number of jobs within that job file. When there are multiple jobs in a FIO job file and a ```global startdelay``` is specified, the FIO metrics collection code expects that ```global startdelay``` is present between each of the jobs in one job file. To make sure the FIO load test has ```global startdelay``` amount of delay between each job, the following ```startdelay``` should be added to the job options in the FIO job file:
+
+    1. For the first job the ```startdelay``` = ```global startdelay```
+
+    2. For every subsequent job, ```startdelay``` = ```startdelay``` of previous job + ```runtime``` of previous job + ```ramp_time``` of previous job +       ```global startdelay```
+
+6. Run the FIO load test
+```bash
+fio job_files/your-job-file.fio --lat_percentiles 1 --output-format=json --output='output.json'
+```
+7. Install requirements by running
+```bash
+pip install -r requirements.txt --user
+```
+8. Create a service account by following this [documentation](https://cloud.google.com/iam/docs/creating-managing-service-accounts). Generate your service account key, `creds.json` by following [this doc](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-console) and upload the file on your GCS bucket `your-bucket-name`. If using an old credentials file, make sure that it is not expired. Run the following command to copy it into `gsheet` directory:
+```bash
+gsutil cp gs://your-bucket-name/creds.json ./gsheet
+```
+9. Create a Google Sheet with id `your-gsheet-id` by copying this [Google Sheet](https://docs.google.com/spreadsheets/d/10fyD6pPyrQoVt9YofT1SyXyjEqPqNhya356ThuA9H_o/edit?usp=sharing).
+10. Share the above copied Google Sheet with your service account(created in step 8)
+11. Change the Google sheet id in this [line](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/perfmetrics/scripts/gsheet/gsheet.py#L5) to `your-gsheet-id`.
```

Comment:
Can this not be made a flag for the script?

Reply:
It cannot be made a flag for the script because we are calling a method of this script from another script